library/src/main/camera2/com/google/android/cameraview/Camera2.java

```diff
@@ -139,17 +142,22 @@ public void onConfigured(@NonNull CameraCaptureSession session) {
             mCaptureSession = session;
             updateAutoFocus();
             updateFlash();
-            if (mStartVideoRecording) {
-                mStartVideoRecording = false;
-                mMediaRecorder.start();
-                mRecording = true;
-            }
-            try {
-                mCaptureSession.setRepeatingRequest(mPreviewRequestBuilder.build(),
-                        mCaptureCallback, null);
-            } catch (CameraAccessException | IllegalStateException e) {
-                Log.e(TAG, "Failed to start camera preview.", e);
-            }
+
+            new Handler(Looper.getMainLooper()).post(new Runnable() {
```

Comment:
We run the callback on a background thread now - but we still need to start mediarecorder on the main thread